### PR TITLE
kaiax/gasless: fix some bugs in Task.ApplyTransactions

### PR DESCRIPTION
**Pull request by Mdaiki0730**
## Proposed changes

This PR fix following bugs in Task.ApplyTransactions.
- `env.tcount` used in `SetTxContext` within `commitBundleTransaction` is not being incremented.
- `numTxsChecked` does not increment correctly for bundles.
- When the return value of `bc.ApplyTransaction` is `err == nil && receipt.Status != ReceiptStatusSuccessful`, `ShiftTxs` is executed instead of `PopTxs`.

## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

_Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/kaiachain/kaia/blob/main/CONTRIBUTING.md) doc
- [x] I have read the [CLA](https://gist.github.com/kaiachain-dev/bbf65cc330275c057463c4c94ce787a6) and signed by comment `I have read the CLA Document and I hereby sign the CLA` in first time contribute
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by hyeonLewis — COMMENTED** [review]
I think https://github.com/kaiachain/kaia/pull/282/files#diff-aa53dfe64c81ad7f4673af6857ca8ae7873bb18224202adb09f765ce4f458411L773 should be updated also.

**Review by ulbqb — APPROVED** [review]

**Comment by ulbqb** [comment]
> EEST is the successor to [ethereum/tests](https://github.com/ethereum/tests) (aka "original tests")

https://github.com/ethereum/execution-spec-tests/pull/1165/files#diff-710a99e33d5a7edcaa8f1464a11316b463987558f100a960a496318389ab0b20

**Comment by hyeonLewis** [comment]
I found one more corner case.

In the `commitBundleTransaction`, it catches `err != nil || receipt.Status != types.ReceiptStatusSuccessful`. It means the `err` can be nil and switch statement recognizes this case as `case nil` so call `ShiftTxs` not `PopTxs`. I think we need to add custom error message for bundle call failure and make a separate case to pop the txs.

**Comment by Mdaiki0730** [comment]
@hyeonLewis 
You're right, I'll change the PR name to broaden the scope of the fix.

**Review by ian0371 — APPROVED** [review]

File: work/worker.go [hunk 1]
```diff
@@ -932,9 +935,14 @@ func (env *Task) commitBundleTransaction(bundle *builder.Bundle, bc BlockChain,
 			}
 			*env.state = *lastSnapshot
 			env.header.GasUsed = gasUsedSnapshot
+			env.tcount = tcountSnapshot
+			if err == nil {
```
Comment (hyeonLewis) [inline work/worker.go:939]: How about ignoring the original `err` and return `kerrors.ErrRevertedBundleByVmErr` always?

Comment (Mdaiki0730) [reply work/worker.go:939]: I think we don't have to override this for proper logging and error counting.
Also, in the case of NonceTooLow, I think it is enough to use shift instead of pop, even if it is a tx in a bundle.
https://github.com/kaiachain/kaia/pull/282/files#diff-aa53dfe64c81ad7f4673af6857ca8ae7873bb18224202adb09f765ce4f458411R812-R816

Comment (hyeonLewis) [reply work/worker.go:939]: Ok, thanks.
